spec/index.bs

```diff
@@ -2036,12 +2043,150 @@ dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
 
 # Pipelines # {#pipelines}
 
+## Base pipelines # {#pipeline-base}
+
 <script type=idl>
 dictionary GPUPipelineDescriptorBase : GPUObjectDescriptorBase {
-    required GPUPipelineLayout layout;
+    GPUPipelineLayout layout;
+};
+
+interface mixin GPUPipelineBase {
+    GPUBindGroupLayout getBindGroupLayout(unsigned long index);
 };
 </script>
 
+{{GPUPipelineBase}} has the following internal slots:
+<dl dfn-type=attribute dfn-for="GPUPipelineBase">
+    : <dfn>\[[layout]]</dfn> of type `GPUPipelineLayout`.
+    ::
+        The definition of the layout of resources which can be used with `this`.
+</dl>
+
+### <dfn method for=GPUPipelineBase>getBindGroupLayout(index)</dfn> ### {#pipelinebase-getBindGroupLayout}
+
+<div algorithm="GPUPipelineBase.getBindGroupLayout">
+    **Arguments:**
+
+        - {{unsigned long}} |index|
+
+    **Returns:** {{GPUBindGroupLayout}}
+
+    1. If |index| is greater or equal to {{GPULimits/maxBindGroups}}:
+
+        1. Throw a {{RangeError}}.
+
+    1. If |this| is not [=valid=]:
+
+        1. Return a new error {{GPUBindGroupLayout}}.
+
+    1. Return a new {{GPUBindGroupLayout}} wrapping
+        |this|.{{GPUPipelineBase/[[layout]]}}.{{GPUPipelineLayout/[[bindGroupLayout]]}}[|index|].
+
+    Note: Only returning new {{GPUBindGroupLayout}} objects ensure no synchronization is necessary
+        between the [=Content timeline=] and the [=Device timeline=].
+
+</div>
+
+### Default pipeline layout ### {#default-pipeline-layout}
+
+A {{GPUPipelineBase}} object that was created without a {{GPUPipelineDescriptorBase/layout}}
+has a default layout created and used instead.
+
+<div algorithm="default pipeline layout creation">
+
+    1. Let |groupDescs| be a sequence of {{GPULimits/maxBindGroups}} new {{GPUBindGroupLayoutDescriptor}} objects.
```

Comment:
for clarity, this needs to say what `GPULimits` this is. We have one on the adapter and one passed into requestDevice